src/kv/kv.h

```diff
@@ -1486,6 +1490,9 @@ namespace kv
       auto h = get_history();
       if (h)
         h->rollback(v);
+      auto e = get_encryptor();
```

Comment:
I wonder if we could/should put the ledger keys in the history and have the encryptor get them from there, and have a single thing we rollback/compact as a result, or if that makes the code worse. Just a thought.

Reply:
Is that because we do not want the kv to know about too many objects that it needs to rollback/compact? 
I think I prefer it as it is since the kv needs to know about the encryptor already and that (hypothetically), a standalone kv could be instantiated without an history but with an encryptor or vice-versa.

Reply:
Yes, it's a complexity tradeoff, I'm not strongly convinced either way, so I'm happy to go with the split.